cmd/create/machinepool/nodepool.go

```diff
@@ -139,8 +139,51 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		}
 	}
 
+	labels := args.labels
+	if interactive.Enabled() {
+		labels, err = interactive.GetString(interactive.Input{
+			Question: "Labels",
+			Help:     cmd.Flags().Lookup("labels").Usage,
+			Default:  labels,
+			Validators: []interactive.Validator{
+				LabelValidator,
+			},
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid comma-separated list of attributes: %s", err)
+			os.Exit(1)
+		}
+	}
+	labelMap, err := ParseLabels(labels)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	taints := args.taints
+	if interactive.Enabled() {
+		taints, err = interactive.GetString(interactive.Input{
+			Question: "Taints",
+			Help:     cmd.Flags().Lookup("taints").Usage,
+			Default:  taints,
+			Validators: []interactive.Validator{
+				taintValidator,
+			},
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid comma-separated list of attributes: %s", err)
+			os.Exit(1)
+		}
+	}
+	taintBuilders, err := parseTaints(taints)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
```

Comment:
surely this code can be exported to functions used both by machine pools and node pools.
There is nothing nodepool-specific here

Reply:
yes makes sense, as I did it already on the edit part

Reply:
refactored